### ISSUE TITLE
Stronger bias from attention_mask in BERT

### ISSUE THREAD

**Issue by fkrasnov2** [huggingface/transformers#34339]
### System Info

all versions

### Who can help?

@Artu

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
# Since attention_mask is 1.0 for positions we want to attend and 0.0 for
# masked positions, this operation will create a tensor which is 0.0 for
# positions we want to attend and -10000.0 for masked positions.
adder = (1.0 - tf.cast(attention_mask, tf.float32)) * -10000.0

# Since we are adding it to the raw scores before the softmax, this is
# effectively the same as removing these entirely.
attention_scores += adder
```

### Expected behavior

Hey there, 

I'm kind of stuck on this attention layer thing and I'm a bit confused about the difference between our implementation and Jacob's original code. It's clear that the attention mask does some biasing before the softmax , but in Jacob's code, it seems like the bias is a lot stronger.

[**Jacob Davlin:**](https://github.com/google-research/bert/blob/eedf5716ce1268e56f0a50264a88cafad334ac61/modeling.py#L712)

![image](https://github.com/user-attachments/assets/050e26b5-501f-40e6-99b7-e43944e084c3)

[**Transformers:**](https://github.com/huggingface/transformers/blob/53fad641cfdb5105e2470bcf3ef17ea8e25cc300/src/transformers/models/bert/modeling_bert.py#L331)

![image](https://github.com/user-attachments/assets/d71e8d76-4177-446f-a28f-9fbf6ae8c8e1)

In the Transformers code, only 1 and 0 are added from attention_mask. 

Maybe I'm wrong. Is it a mistake? What do you think?

Best, 
Fedor

**Comment by vasqu** [comment]
Hey Fedor, the attention mask preperation happens elsewhere and depends on the attention implementation (eager, flash attn, sdpa). The only difference to the original version here in Bert is that it uses torch's min dtype instead of the hardcoded -10000.

See the attention preparation here (the last line is for eager for example):
https://github.com/huggingface/transformers/blob/53fad641cfdb5105e2470bcf3ef17ea8e25cc300/src/transformers/models/bert/modeling_bert.py#L1086-L1114
and the inversion in eager you're looking for is over here:
https://github.com/huggingface/transformers/blob/343c8cb86f2ab6a51e7363ee11f69afb1c9e839e/src/transformers/modeling_utils.py#L1188
